pkg/util/log/zapencoder.go

```diff
@@ -0,0 +1,124 @@
+	enc := &sliceArrayEncoder{elems: make([]interface{}, 0)}
+	err := v.MarshalLogArray(enc)
+	e.ctx = append(e.ctx, e.fullKey(k), enc.elems)
+	return err
+}
+
+func (e *encoder) AddObject(k string, v zapcore.ObjectMarshaler) error { return v.MarshalLogObject(e) }
+func (e *encoder) AddBinary(k string, v []byte)                        { e.ctx = append(e.ctx, e.fullKey(k), v) }
+func (e *encoder) AddByteString(k string, v []byte)                    { e.ctx = append(e.ctx, e.fullKey(k), string(v)) }
+func (e *encoder) AddBool(k string, v bool)                            { e.ctx = append(e.ctx, e.fullKey(k), v) }
+func (e *encoder) AddDuration(k string, v time.Duration)               { e.ctx = append(e.ctx, e.fullKey(k), v) }
+func (e *encoder) AddComplex128(k string, v complex128)                { e.ctx = append(e.ctx, e.fullKey(k), v) }
+func (e *encoder) AddComplex64(k string, v complex64)                  { e.ctx = append(e.ctx, e.fullKey(k), v) }
+func (e *encoder) AddFloat64(k string, v float64)                      { e.ctx = append(e.ctx, e.fullKey(k), v) }
+func (e *encoder) AddFloat32(k string, v float32)                      { e.ctx = append(e.ctx, e.fullKey(k), v) }
+func (e *encoder) AddInt(k string, v int)                              { e.ctx = append(e.ctx, e.fullKey(k), v) }
+func (e *encoder) AddInt64(k string, v int64)                          { e.ctx = append(e.ctx, e.fullKey(k), v) }
+func (e *encoder) AddInt32(k string, v int32)                          { e.ctx = append(e.ctx, e.fullKey(k), v) }
+func (e *encoder) AddInt16(k string, v int16)                          { e.ctx = append(e.ctx, e.fullKey(k), v) }
+func (e *encoder) AddInt8(k string, v int8)                            { e.ctx = append(e.ctx, e.fullKey(k), v) }
+func (e *encoder) AddString(k string, v string)                        { e.ctx = append(e.ctx, e.fullKey(k), v) }
+func (e *encoder) AddTime(k string, v time.Time)                       { e.ctx = append(e.ctx, e.fullKey(k), v) }
+func (e *encoder) AddUint(k string, v uint)                            { e.ctx = append(e.ctx, e.fullKey(k), v) }
+func (e *encoder) AddUint64(k string, v uint64)                        { e.ctx = append(e.ctx, e.fullKey(k), v) }
+func (e *encoder) AddUint32(k string, v uint32)                        { e.ctx = append(e.ctx, e.fullKey(k), v) }
+func (e *encoder) AddUint16(k string, v uint16)                        { e.ctx = append(e.ctx, e.fullKey(k), v) }
+func (e *encoder) AddUint8(k string, v uint8)                          { e.ctx = append(e.ctx, e.fullKey(k), v) }
+func (e *encoder) AddUintptr(k string, v uintptr)                      { e.ctx = append(e.ctx, e.fullKey(k), v) }
+func (e *encoder) AddReflected(k string, v interface{}) error {
+	e.ctx = append(e.ctx, e.fullKey(k), v)
+	return nil
+}
+
+func (e *encoder) OpenNamespace(ns string) {
+	if e.prefix == "" {
+		e.prefix = ns + "/"
+	} else {
+		e.prefix = e.prefix + ns + "/"
+	}
+}
```

Comment:
Would the following be equivalent to the current implementation:
```suggestion
func (e *encoder) OpenNamespace(ns string) {
	e.prefix = e.prefix + ns + "/"
}
```

given that `e.prefix + ns == ns` if `e.prefix == ""`?